cabal-benchmark.project

```diff
@@ -0,0 +1,59 @@
+packages:
+         ./
+         ./hie-compat
+         ./shake-bench
+         ./hls-graph
+         ./ghcide
+         ./hls-plugin-api
+         ./hls-test-utils
+         ./plugins/hls-tactics-plugin
+         ./plugins/hls-brittany-plugin
+         ./plugins/hls-stylish-haskell-plugin
+         ./plugins/hls-fourmolu-plugin
+         ./plugins/hls-class-plugin
+         ./plugins/hls-eval-plugin
+         ./plugins/hls-explicit-imports-plugin
+         ./plugins/hls-refine-imports-plugin
+         ./plugins/hls-hlint-plugin
+         ./plugins/hls-retrie-plugin
+         ./plugins/hls-haddock-comments-plugin
+         ./plugins/hls-splice-plugin
+         ./plugins/hls-floskell-plugin
+         ./plugins/hls-pragmas-plugin
+         ./plugins/hls-module-name-plugin
+         ./plugins/hls-ormolu-plugin
+         ./plugins/hls-call-hierarchy-plugin
+
+benchmarks: true
+
+package *
+  ghc-options: -haddock
+  test-show-details: direct
+
+-- see https://gitlab.haskell.org/haskell/ghc-api-compat/-/issues/1#note_371055
+source-repository-package
+  type: git
+  location: https://github.com/hsyl20/ghc-api-compat
+  tag: 8fee87eac97a538dbe81ff1ab18cff10f2f9fa15
+
+write-ghc-environment-files: never
+
+index-state: 2021-08-12T12:00:38Z
+
+constraints:
+    hyphenation +embed
+
+allow-newer:
+    diagrams-contrib:base,
+    diagrams-contrib:lens,
+    diagrams-contrib:random,
+    diagrams-core:base,
+    diagrams-core:lens,
+    diagrams-lib:base,
+    diagrams-lib:lens,
+    diagrams-postscript:base,
+    diagrams-postscript:lens,
+    diagrams-svg:base,
+    diagrams-svg:lens,
+    dual-tree:base,
+    svg-builder:base
```

Comment:
Are these still needed though? The upstream issue was fixed recently: https://github.com/diagrams/diagrams-core/issues/112

Reply:
will try it

Reply:
without the allow-newer for diagrams-core i got

```
PS D:\hls> cabal bench --project-file .\cabal-benchmark.project
Resolving dependencies...
cabal-3.4.0.0.exe: Could not resolve dependencies:
[__0] trying: ghc-api-compat-8.6.1 (user goal)
[__1] trying: base-4.14.2.0/installed-4.14.2.0 (dependency of ghc-api-compat)
[__2] trying: shake-bench-0.1.0.0 (user goal)
[__3] trying: diagrams-svg-1.4.3 (dependency of shake-bench)
[__4] trying: diagrams-core-1.5.0 (dependency of diagrams-svg)
[__5] next goal: Chart-diagrams (dependency of shake-bench)
[__5] rejecting: Chart-diagrams-1.9.3 (conflict: diagrams-core==1.5.0,
Chart-diagrams => diagrams-core>=1.3 && <1.5)
[__5] skipping: Chart-diagrams-1.9.2, Chart-diagrams-1.9.1,
Chart-diagrams-1.9, Chart-diagrams-1.8.3, Chart-diagrams-1.8.2,
Chart-diagrams-1.8.1, Chart-diagrams-1.8, Chart-diagrams-1.7.1,
Chart-diagrams-1.7, Chart-diagrams-1.6, Chart-diagrams-1.5.4,
Chart-diagrams-1.5.1, Chart-diagrams-1.5, Chart-diagrams-1.4,
Chart-diagrams-1.3.3, Chart-diagrams-1.3.2, Chart-diagrams-1.3.1,
Chart-diagrams-1.3, Chart-diagrams-1.2.4, Chart-diagrams-1.2.3,
Chart-diagrams-1.2.2, Chart-diagrams-1.2, Chart-diagrams-1.1,
Chart-diagrams-1.0 (has the same characteristics that caused the previous
version to fail: excludes 'diagrams-core' version 1.5.0)
[__5] fail (backjumping, conflict set: Chart-diagrams, diagrams-core,
shake-bench)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: diagrams-core, base, Chart-diagrams,
shake-bench, diagrams-svg, ghc-api-compat
Try running with --minimize-conflict-set to improve the error message.
```

After trying several combinations i only get it to build enabling all of them maybe a more detailed analysis could make let us know the way to remove them.

Reply:
I've got to remove some other entries from `allow-newer`